update legacy dispatcher registration API tests to avoid duplicate def() calls

bdhirsh opened the pull request:
The tests changed in this diff all have duplicate `def()` calls through the dispatcher API, due to using the legacy registration API. I tried to clean them up in a sane way:

(a) Fixed it, if possible, if the test semantically tested the same thing (still using the legacy API)
(b) migrated it to use the new registration API (I did this in tests for which I couldn't figure out a way to get the legacy API to correctly express the test's intent without causing a duplicate def()).
(c) Deleted it (I only did this for tests in op_registration_test. Mostly tests that seemed to be testing the semantic behavior of using duplicate def()'s).

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47322 make duplicate def() calls an error in the dispatcher
* #47321 migrate export_caffe2_op_to_c10.h macros to the new dispatcher registration API
* #47320 rename macro. TORCH_LIBRARY_FRAGMENT_THIS_API_IS_FOR_PER_OP_REGISTRATION_ONLY to TORCH_LIBRARY_FRAGMENT
* **#47319 update legacy dispatcher registration API tests to avoid duplicate def() calls**
* #47318 remove ops in the __caffe2 namespace

Differential Revision: [D24714804](https://our.internmc.facebook.com/intern/diff/D24714804)

bdhirsh committed:
update legacy dispatcher registration API tests to avoid duplicate def() calls

[ghstack-poisoned]

bdhirsh committed:
Update on "update legacy dispatcher registration API tests to avoid duplicate def() calls"

The tests changed in this diff all have duplicate `def()` calls through the dispatcher API, due to using the legacy registration API. I tried to clean them up in a sane way:

(a) Fixed it, if possible, if the test semantically tested the same thing (still using the legacy API)
(b) migrated it to use the new registration API (I did this in tests for which I couldn't figure out a way to get the legacy API to correctly express the test's intent without causing a duplicate def()).
(c) Deleted it (I only did this for tests in op_registration_test. Mostly tests that seemed to be testing the semantic behavior of using duplicate def()'s).


Differential Revision: [D24714804](https://our.internmc.facebook.com/intern/diff/D24714804)

[ghstack-poisoned]

ezyang commented:
added @smessmer here for visibility (Brian, you should feel free to start the landing process)

bdhirsh committed:
Update on "update legacy dispatcher registration API tests to avoid duplicate def() calls"

The tests changed in this diff all have duplicate `def()` calls through the dispatcher API, due to using the legacy registration API. I tried to clean them up in a sane way:

(a) Fixed it, if possible, if the test semantically tested the same thing (still using the legacy API)
(b) migrated it to use the new registration API (I did this in tests for which I couldn't figure out a way to get the legacy API to correctly express the test's intent without causing a duplicate def()).
(c) Deleted it (I only did this for tests in op_registration_test. Mostly tests that seemed to be testing the semantic behavior of using duplicate def()'s).


Differential Revision: [D24714804](https://our.internmc.facebook.com/intern/diff/D24714804)

[ghstack-poisoned]
